### PR TITLE
Homework 1

### PR DESCRIPTION
**Pull request by OGrabar**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,6 +8,70 @@
...
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <calculator-version>1.0</calculator-version>
+        <testng-version>7.4.0</testng-version>
+    </properties>
 
+    <dependencies>
+        <dependency>
+            <groupId>com.epam</groupId>
```
Comment (HronoSF) [inline pom.xml:20]: You forget to commit .jar file :)
image

File: src/test/java/hw1/data_providers/ArithmeticDataProviders.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+package hw1.data_providers;
+
+import org.testng.annotations.DataProvider;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+public class ArithmeticDataProviders {
```
Comment (HronoSF) [inline src/test/java/hw1/data_providers/ArithmeticDataProviders.java:7]: ThreadLocalRandom.current() should be extracted to class field

File: src/test/java/hw1/CalculatorBaseTest.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import org.testng.annotations.BeforeClass;
+
+public class CalculatorBaseTest {
+
+    protected Calculator calculator;
+    protected static final double DOUBLE_DELTA = 1e-6;
+
+    @BeforeClass
+    public void setUp() {
+        System.out.println("213");
```
Comment (HronoSF) [inline src/test/java/hw1/CalculatorBaseTest.java]: console log?

**Review by HronoSF — COMMENTED** [review]
Greate job!
Code looks good.
I will wait for .jar to run your tests

File: src/test/java/hw1/CalculatorBaseTest.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import com.epam.tat.module4.Calculator;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+
+public class CalculatorBaseTest {
+
+    protected Calculator calculator;
+    protected static final double DOUBLE_DELTA = 1e-6;
+
+    @BeforeClass
```
Comment (HronoSF) [inline src/test/java/hw1/CalculatorBaseTest.java:11]: why did you decide to use @BeforeClass, @AfterClass?

File: src/test/java/hw1/CalculatorBaseTest.java (hunk 3 above)
Comment (OGrabar) [reply src/test/java/hw1/CalculatorBaseTest.java]: Yep. Forget to delete this one. Removed now

File: src/test/java/hw1/data_providers/ArithmeticDataProviders.java (hunk 2 above)
Comment (OGrabar) [reply src/test/java/hw1/data_providers/ArithmeticDataProviders.java:7]: Thanks. Don't think about it. Fixed

File: src/test/java/hw1/CalculatorBaseTest.java (hunk 4 above)
Comment (OGrabar) [reply src/test/java/hw1/CalculatorBaseTest.java:12]: I think that it's a bit redundant to init new Calculator before each test. So I tried to decide what's better @BeforeClass or @BeforeSuite. But in this tests there are no difference beetwen annotations. Now changed to @BeforeSuite

File: pom.xml (hunk 1 above)
Comment (OGrabar) [reply pom.xml:20]: Lol. So stupid mistake. Fixed

Comment (OGrabar) [reply pom.xml:20]: Some of test cases should fail. There are some bugs in Calculator class.

File: src/test/java/hw1/CalculatorBaseTest.java (hunk 4 above)
Comment (OGrabar) [reply src/test/java/hw1/CalculatorBaseTest.java:11]: Because I think that @BeforeTest is a bit redundant

**Pull request closed without merging**